tdwii_plus_examples/ngetscu.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import sys
+from typing import Iterator, List
+
+import pydicom
+import pynetdicom
+from pydicom import DataElement, Dataset
+from pynetdicom import UnifiedProcedurePresentationContexts
+from pynetdicom.sop_class import UnifiedProcedureStepPush
+
+import tdwii_plus_examples.tdwii_config as ae_config
+
+
+def nget_ups(
+    attribute_list: List[DataElement], ups_instance_uid: str, scu_ae_title: str, scp_ae_title: str
+) -> Iterator[tuple[Dataset, Dataset | None]]:
+    """Perform UPS N-GET-RQ and return responses that have UPS content
+
+    Args:
+        ds_query (pydicom.Dataset): query content/request
+        scu_ae_title (str): The AE Title for the SCU (the requestor), typically the PPVS or TDS
+        scp_ae_title (str): The AE Title for the SCP (being queried), the TMS
+
+    Yields:
+        Iterator[tuple[Dataset, Dataset | None]]: status,response_content tuples
+    """
+    ae = pynetdicom.AE(ae_title=scu_ae_title)
+    ae.requested_contexts = UnifiedProcedurePresentationContexts
+    query_model = UnifiedProcedureStepPush
+    ups_responses = list()
+
+    if scp_ae_title not in ae_config.known_ae_ipaddr:
+        logging.error("Unable to query " + scp_ae_title + " not found in this TDW-II Application Entity configuration")
+        return
+    identifier_list = set([x.tag for x in attribute_list])
+    # SOP Class UID, SOP Instance UID, Transaction UID are not allowed
+    disallowed_tags = set([0x00080016, 0x00080018, 0x00081195])
+    identifier_list.difference_update(disallowed_tags)
+    for tag in identifier_list:
+        print(tag)
```

Comment:
may be modified to logging.info

Reply:
fixed in 0e0d99882130b7aee1bd5f0ae0b6df0193e7b4e2